File: 2483. Minimum Penalty for a Shop/js/solution.js
/**
 * @param {string} customers
 * @return {number}
 */
var bestClosingTime = function (customers) {
    let pen = 0;
    let minPen = 0;
    let idx = 0;

    for (let i = 0; i < customers.length; i++) {
        const s = customers[i];

        if (s === 'Y') pen--;
        if (s === 'N') pen++;

        if (pen < minPen) {
            idx = i + 1;
            minPen = pen;
        }
    }

    return idx;
};

File: 2483. Minimum Penalty for a Shop/cs/solution.cs
public class Solution {
    public int BestClosingTime(string customers) {
        int pen = 0;
        int minPen = 0;
        int idx = 0;

        for(int i = 0; i < customers.Length; i++) {
            char s = customers[i];
            if (s == 'Y') pen--;
            if (s == 'N') pen++;

            if (pen < minPen) {
                idx = i + 1;
                minPen = pen;
            }
        }

        return idx;
    }
}